src/bin/sol-fbp-generator/main.c

```diff
@@ -694,8 +727,20 @@ main(int argc, char *argv[])
         goto fail_resolve;
     }
 
+    fd = open(".", O_TMPFILE | O_RDWR, S_IRUSR | S_IWUSR);
```

Comment:
The generator is a pretty portable piece of software, and `O_TMPFILE` is not only Linux-specific, it's also pretty recent (3.11).

Also, I'm not sure how opening "." is the way to go. What if the process' current working directory sits on a read-only file system? What if the current directory is on a different filesystem than `args.output_file`... will `linkat()` still work?